src/robot/running/testlibraries.py

```diff
@@ -182,7 +183,7 @@ def _create_handlers(self, libcode):
             if method:
                 handler = self._try_to_create_handler(name, method)
                 if handler:
-                    self.handlers.add(handler)
+                    self.handlers.add(handler, handler.__class__.__name__ == 'EmbeddedArgsTemplate')
```

Comment:
1. Wouldn't `isinstance(handler, EmbeddedArgsTemplate) ` work here instead of checking class name?

2. It would be a little cleaner to either to pass the Boolean argument with `embedded=` prefix or alternatively to extract a local variable containing the value.